crates/ruma-common/src/events/call/hangup.rs

```diff
@@ -2,35 +2,81 @@
 //!
 //! [`m.call.hangup`]: https://spec.matrix.org/v1.2/client-server-api/#mcallhangup
 
-use js_int::UInt;
 use ruma_macros::EventContent;
 use serde::{Deserialize, Serialize};
 
-use crate::{serde::StringEnum, PrivOwnedStr};
+use crate::{serde::StringEnum, OwnedVoipId, PrivOwnedStr, VoipVersionId};
 
 /// The content of an `m.call.hangup` event.
 ///
-/// Sent by either party to signal their termination of the call. This can be sent either once the
-/// call has has been established or before to abort the call.
+/// Sent by either party to signal their termination of the call.
+///
+/// In VoIP version 0, this can be sent either once the call has been established or before to abort
+/// the call.
+///
+/// With the `unstable-msc2746` feature, and if the call is using VoIP version 1, this should only
+/// be sent by the caller after sending the invite or by the callee after answering the invite. To
+/// reject an invite, send an [`m.call.reject`] event.
+///
+/// [`m.call.reject`]: super::reject::CallRejectEventContent
 #[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
 #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
 #[ruma_event(type = "m.call.hangup", kind = MessageLike)]
 pub struct CallHangupEventContent {
-    /// The ID of the call this event relates to.
-    pub call_id: String,
+    /// A unique identifier for the call.
+    pub call_id: OwnedVoipId,
+
+    #[cfg(feature = "unstable-msc2746")]
+    /// **Required in VoIP version 1.** A unique ID for this session for the duration of the call.
+    ///
+    /// Must be the same as the one sent by the previous invite or answer from
+    /// this session.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub party_id: Option<OwnedVoipId>,
 
     /// The version of the VoIP specification this messages adheres to.
-    pub version: UInt,
+    pub version: VoipVersionId,
 
     /// Optional error reason for the hangup.
+    ///
+    /// With the `unstable-msc2746` feature, this field is required.
+    #[cfg(not(feature = "unstable-msc2746"))]
     #[serde(skip_serializing_if = "Option::is_none")]
     pub reason: Option<Reason>,
+
+    /// Error reason for the hangup.
+    ///
+    /// Defaults to [`Reason::UserHangup`] if it is missing.
+    ///
+    /// Without the `unstable-msc2746` feature, this field is optional.
+    #[cfg(feature = "unstable-msc2746")]
+    #[serde(default)]
+    pub reason: Reason,
```

Comment:
This is non-additive too. The issue is that the default value doesn't exist without the feature flag.

Should I just leave it as an `Option` that is always `Some` with the unstable feature? Then we can remove the `Option` when it is stabilized.

Reply:
That sounds reasonable **if** the default is not supposed to be scoped to just version 1 of this feature.